core/src/main/java/org/apache/brooklyn/core/entity/AbstractEntity.java

```diff
@@ -1698,6 +1698,7 @@ public void add(Policy policy) {
             CatalogUtils.setCatalogItemIdOnAddition(AbstractEntity.this, policy);
             policiesInternal.add((AbstractPolicy)policy);
             ((AbstractPolicy)policy).setEntity(AbstractEntity.this);
+            ConfigConstraints.assertValid(policy);
```

Comment:
Seems strange to add + activate the policy, and only then assert that it is valid. But I guess it might not be fully initialised before `setEntity` is called, for us to validate it?

Hopefully the policy was instantiated using the normal mechanism (so went through `InternalPolicyFactory.createPolicy`, which will have called `assertValid`.

Are there situations where this code catches something we were previously missing? If so, should we write a **test** for it?